Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py

```diff
@@ -0,0 +1,192 @@
+    """
+    def __init__(self, client: Client) -> None:
+        self.client = client
+
+    def _iter_events(self) -> Generator:
+        """
+        Function that responsible for the iteration over the events returned from github api
+        """
+        response = self.client.call()
+        events: list = response.json()
+
+        if not events:
+            return []
+
+        while True:
+            yield events
+            last = events.pop()
+            self.client.set_next_run_filter(last['@timestamp'])
+            response = self.client.call()
+            events = response.json()
+            try:
+                events.pop(0)
+                assert events
+            except (IndexError, AssertionError):
+                LOG('empty list, breaking')
+                break
+
+    def aggregated_results(self, limit=2000) -> List[dict]:
+        """
+        Function to group the events returned from the api
+        """
+        stored_events = []
+        for events in self._iter_events():
+            stored_events.extend(events)
+            if len(stored_events) >= limit:
+                return stored_events[:limit]
+        return stored_events
+
+    @staticmethod
+    def get_last_run(events: List[dict]) -> dict:
+        """
+        Get the info from the last run, it returns the time to query from and a list of ids to prevent duplications
+        """
+
+        last_timestamp = events[-1]['@timestamp']
+        last_time = last_timestamp / 1000
+        next_fetch_time = datetime.fromtimestamp(last_time) + timedelta(
+            seconds=1
+        )
+        return {'after': next_fetch_time.isoformat()}
+
+
+def main():
+    # Args is always stronger. Get last run even stronger
+    demisto_params = demisto.params() | demisto.args() | demisto.getLastRun()
+    events_to_add_per_request = demisto_params.get('events_to_add_per_request', 1000)
+    try:
+        events_to_add_per_request = int(events_to_add_per_request)
+    except ValueError:
+        events_to_add_per_request = 1000
+
+    headers = {'Authorization': f"Bearer {demisto_params['auth_credendtials']['password']}",
+               'Accept': 'application/vnd.github.v3+json'}
+
+    demisto_params['headers'] = headers
+    demisto_params['params'] = ReqParams(**demisto_params)
+
+    request = Request(**demisto_params)
+
+    client = Client(request)
+
+    get_events = GetEvents(client)
+
+    command = demisto.command()
+    try:
+        urllib3.disable_warnings()
+
+        if command == 'test-module':
+            get_events.aggregated_results(limit=1)
+            return_results('ok')
+        elif command == 'github-get-events' or command == 'fetch-events':
+            events = get_events.aggregated_results(limit=int(demisto_params.get('limit')))
+
+            if command == 'fetch-events':
+                if events:
+                    demisto.setLastRun(GetEvents.get_last_run(events))
+                while len(events) > 0:
```

Comment:
This should be `>=`, the send_events_to_xsiam function also handles empty events case.